src/vs/platform/notification/common/notification.ts

```diff
@@ -329,6 +334,11 @@ export interface INotificationService {
 	 */
 	readonly onDidRemoveNotification: Event<INotification>;
 
+	/**
+ * Emitted when a do not disturb mode has changed.
+ */
+	readonly onDidSetDoNotDisturbMode: Event<NotificationsDoNotDisturbMode>;
```

Comment:
I suggest `onDidChangeDoNotDisturbMode: Event<void>` because figuring out if DND mode is enabled or not is an easy call from the service.